server/handlers/validation.go

```diff
@@ -28,9 +30,13 @@ func validateUpdate(cs signed.CryptoService, gun string, updates []storage.MetaU
 	kdb := keys.NewDB()
 	repo := tuf.NewRepo(kdb, cs)
 	rootRole := data.RoleName(data.CanonicalRootRole)
-	targetsRole := data.RoleName(data.CanonicalTargetsRole)
 	snapshotRole := data.RoleName(data.CanonicalSnapshotRole)
 
+	// some delegated targets role may be invalid based on other updates
+	// that have been made by other clients. We'll rebuild the slice of
+	// updates with only the things we should actually update
+	updatesToApply := make([]storage.MetaUpdate, 0, len(updates))
+
 	// check that the necessary roles are present:
```

Comment:
Non-blocking - sorry, this was not part of your change at all, but this particular comment seems no longer applicable.